### PR TITLE
Improve output file handling

### PR DESCRIPTION
**Pull request by fmeum**

### CODE REVIEW

**Review by yugr — APPROVED** [review]

File: src/hider.c [hunk 1]
```diff
@@ -124,15 +124,15 @@ void hide_symbols(const char *file, const char *out_file,
 
   // Write results
 
-  int out_fd = open(out_file, O_WRONLY);
+  int out_fd = open(out_file, O_WRONLY|O_CREAT);
```
Comment (yugr) [inline src/hider.c]: Could you change to `O_WRONLY | O_CREAT`? Also please check the failing tests - it seems we now need to add the `mode` parameter.

Thanks for the PR!

Comment (fmeum) [reply src/hider.c]: Good point, changed.

I'm planning to use `SymbolHider` as a build step over at https://github.com/CodeIntelligenceTesting/jazzer, but for that require a stable way to refer to a source archive. Could you perhaps push a tag (no need for a release) after this has been merged? See https://github.com/bazel-contrib/SIG-rules-authors/issues/11#issuecomment-1029861300 for why that's needed.

Comment (yugr) [reply src/hider.c]: > I'm planning to use SymbolHider as a build step over at https://github.com/CodeIntelligenceTesting/jazzer

Glad to hear! Let me know if you encounter any issues.

> Could you perhaps push a tag (no need for a release) after this has been merged?

Will do (later today).

**Comment by yugr** [comment]
I've added a v1.0 tag.

**Pull request merged**